### PR TITLE
Cal 79 drag event creation

### PR DESCRIPTION
**Pull request by sofita**
_Closes:_ #79 

## Summary of changes
Add ability to click and drag over multiple days/times to create an event

## Testing

- [x] Unit tests
image
image

### CODE REVIEW

File: ui/src/App.test.tsx [hunk 1]
```diff
@@ -266,12 +268,12 @@ describe("App", () => {
...
       });
       expect(
-        screen.getByText("Error: end cannot be before start.")
+        screen.getByText("Error: end cannot be before start."),
       ).toBeInTheDocument();
     });
   });
 
-  describe("errors", () => {
+  describe("DB interaction errors", () => {
```
Comment (sofita) [inline ui/src/App.test.tsx:276]:
```suggestion
  describe("Backend interaction errors", () => {
```

Comment (sofita) [inline ui/src/App.test.tsx:276]:
```suggestion
  describe("Backend interaction errors", () => {
```

File: ui/src/App.tsx [hunk 2]
```diff
@@ -73,20 +73,22 @@ const App = () => {
...
   const [events, setEvents] = useState<EventSourceInput>([]);
   const [displayedEventData, setDisplayedEventData] = useState(
-    {} as DisplayedEventData
+    {} as DisplayedEventData,
   );
   const [showOverlay, setShowOverlay] = useState<boolean>(false);
   const [inEditMode, setInEditMode] = useState<boolean>(false);
   const [inCreateMode, setInCreateMode] = useState<boolean>(false);
   const [modalDate, setModalDate] = useState<string>(DEFAULT_DATE);
+  const [modalStartDate, setModalStartDate] = useState<string>("");
```
Comment (sofita) [inline ui/src/App.tsx:90]: can we simplify to not have `modalDate` anymore?

File: ui/src/App.tsx [hunk 3]
```diff
@@ -254,24 +256,25 @@ const App = () => {
...
-                  setModalEndTime(
-                    oneHourLater(DateClickObject.date.toUTCString())
-                  );
-                  setModalAllDay(DateClickObject.allDay);
+                select={({ start, end, allDay }) => {
+                  setModalAllDay(allDay);
+
+                  setModalStartDate(formatDate(start));
+                  if (allDay) {
+                    setModalEndDate(formatEndDate(end));
```
Comment (sofita) [inline ui/src/App.tsx:264]:
```suggestion
                    setModalEndDate(formatDateMinusOneDay(end));
```

File: ui/src/App.tsx [hunk 4]
```diff
@@ -342,8 +345,10 @@ const App = () => {
                   <EventForm
                     initialTitle=""
                     initialDescription=""
-                    initialStartDate={modalDate}
-                    initialEndDate={modalDate}
+                    initialStartDate={
+                      modalStartDate ? modalStartDate : modalDate
+                    }
+                    initialEndDate={modalEndDate ? modalEndDate : modalDate}
```
Comment (sofita) [inline ui/src/App.tsx:351]: Can be simplified once `modalDate` is removed

File: ui/src/lib.ts [hunk 5]
```diff
@@ -12,6 +12,16 @@ const formatDate: Function = (date: Date): string => {
   ].join("-");
 };
 
+const formatEndDate: Function = (date: Date): string => {
```
Comment (sofita) [inline ui/src/lib.ts:15]: rename accordingly

File: ui/src/App.tsx (hunk 2 above)
Comment (angepol) [reply ui/src/App.tsx:95]: Yes, I just removed this state variable we can use modalStartDate and modalEndDate instead.

**Review by elliedori — APPROVED** [review]
👍

**Pull request merged**